### PR TITLE
Add swagger

### PR DESCRIPTION
**Pull request by toloco**
Using SwaggerUI from:  https://github.com/swagger-api/swagger-ui (Apache license)

Screenshot 2022-11-03 at 14 40 54

### CODE REVIEW

File: svc-src/nxsearch_svc.lua [hunk 1]
```diff
@@ -158,11 +205,27 @@ routes:delete("@/:string/remove/:number", function(self, name, doc_id)
...
+              schema:
+                $ref: "#/components/error_response"
+  --]]
+
   local index = get_nxs_index(name)
   local query_string = ngx.req.get_uri_args()
   local params = query_string_to_params(query_string)
 
-  local resp, err = index:search(get_http_body(true), params)
+  local resp, err = index:search(get_http_bodyc(true), params)
```
Comment (rmind) [inline svc-src/nxsearch_svc.lua:228]: Typo?

File: compose/docs.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta content="NXSearch docs"/>
```
Comment (rmind) [inline compose/docs.html:6]: s/NXSearch/nxsearch/

File: tools/docs_parser.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        for code, content in yml.get("responses", {}).items()
+    }
+
+    description = yml.get("description", ""),
+
+    return description, request_body, responses
+
+def parse(input: str, output: str):
+    """
+    Get swagger inline: https://github.com/readmeio/swagger-inline
```
Comment (rmind) [inline tools/docs_parser.py:55]: Hmm.. can't we just use off-the-shelf swagger-inline then?

Comment (rmind) [reply tools/docs_parser.py:55]: If it doesn't support Lua comments, then we could just adjust the script to extract them and replae with JS-style comments. The simpler it is on our side, the better.

File: compose/docs.html (hunk 2 above)
Comment (toloco) [reply compose/docs.html:6]:
```suggestion
    <meta content="nxsearch docs"/>
```

File: tools/docs_parser.py (hunk 3 above)
Comment (toloco) [reply tools/docs_parser.py:55]: I couldnt find anything with lua sadly, and this one is so convoluted that Ive endup going with python.
Also we could just go with generating manually, actually seen the comments, we could keep it in the same file kinda how it is now, and only extract the comments as big yaml file

Comment (rmind) [reply tools/docs_parser.py:55]: I was thinking about: 1) prefix the comments with some special characters (e.g. four dashes); 2) `grep` them out forming a new file; 3) pass to a Swagger generator.

**Review by rmind — APPROVED** [review]

**Pull request merged**